relational_operators/AggregationOperator.hpp

```diff
@@ -103,23 +109,23 @@ class AggregationWorkOrder : public WorkOrder {
    * @brief Constructor
    *
    * @param input_block_id The block id.
-   * @param aggr_state_index The index of the AggregationState in QueryContext.
+   * @param state The AggregationState to use.
    **/
   AggregationWorkOrder(
       const block_id input_block_id,
-      const QueryContext::aggregation_state_id aggr_state_index)
+      AggregationOperationState *state)
```

Comment:
Alright. Then this PR is specific to single node, right? If that's the case, I think it looks good @pateljm.